Mannequin

DragonSlayer62 opened the pull request:
- [ADD] Added placeable Mannequin NPCs via deed. Players can equip mannequins with items, swap clothing with them, customize race and gender, and add a custom description

DragonSlayer62 committed:
Mannequin

- [ADD] Added the Mannequin npcs players can place using a deed.

DragonSlayer62 committed:
cleanup

DragonSlayer62 committed:
fixed the packet

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Update mannequin.js

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
update

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Update ItemMenu.bulk.dfn

DragonSlayer62 committed:
cleanup

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

Xoduz commented on source/cPlayerAction.cpp:
```diff
@@ -663,9 +663,18 @@ bool CPIEquipItem::Handle( void )
+		TAGMAPOBJECT tOwner = k->GetTag( "MannequinOwner" );
+		SERIAL ownerSer     = INVALIDSERIAL;
+
+		bool isOwner = ( ownerSer != INVALIDSERIAL && ownerSer == ourChar->GetSerial() ) || ( k->GetOwnerObj() == ourChar );
```
ownerSer is always equal to INVALIDSERIAL (it's set once on the line above, but never changed after that), so isOwner will never be true.

Xoduz commented on data/js/server/network/helpers/openpaperdoll.js:
```diff
@@ -0,0 +1,14 @@
+    pStream.ReserveSize( 66 );
+    pStream.WriteByte( 0, 0x88 );
+    pStream.WriteLong( 1, targChar.serial );
+    pStream.WriteString( 5, targChar.name + ( targChar.title ? " " + targChar.title : "" ), targChar.name.length + ( targChar.title ? targChar.title.length + 1 : 0 ));
```
This needs to respect uox.ini settings for what to show in paperdolls, whether it's for a player or an NPC. Basically, to be a useful helper function, it needs to replicate the functionality provided via the following function in cPlayerAction.cpp:

> void PaperDoll( CSocket *s, CChar *pdoll )

Note that this helper function also will only work for characters that are within client's update range, and may or may not require a .Refresh( socket ) to work properly (see wholist.js script).

Alternatively, if you need this just for the mannequin, include this in the mannequin script instead of a separate helper script. Custom packets can be sent in regular scripts, they don't need to be "network scripts" for that.

DragonSlayer62 replied:
I'll just add it to the mannequin it's not needed for anything more

DragonSlayer62 replied:
removed openpaperdoll  just added to the manny

Xoduz commented on data/js/server/misc/helpers/profanitycheck.js:
```diff
@@ -0,0 +1,148 @@
+// Baseline single-word profanity (no slurs here; keep repo-friendly).
+// Add/remove to suit your shard's policy.
+let bad_words = [
```
Yeah, let's not do this. Make this read from an optional file of banned words instead (one per line), that shard admins can create or provide, but let's not host that (or a hard-defined list of profanities) in the UOX3 repo.

DragonSlayer62 replied:
its more keep people from setting things on the manny but ill just remove it keep it as my own .

Xoduz commented on data/js/jse_fileassociations.scp:
```diff
@@ -388,6 +389,15 @@
+18000=server/network/helpers/openpaperdoll.js
+18001=server/network/helpers/modifycontextmenu.js
+18002=server/misc/helpers/profanitycheck.js
+}
```
There seems to be an extra } here than there's supposed to be.

DragonSlayer62 replied:
okay remove the profanity check did normal description check and move this this to manny since only used on him.

Xoduz commented on data/dfndata/npc/malevendors.dfn:
```diff
@@ -782,6 +782,23 @@ NPCWANDER=0
 GET=m_playervendor f_playervendor
 }
 
+[mannequin]
```
Why is a mannequin a vendor?

DragonSlayer62 replied:
because I just used the player vendor settings to create it so acts like its owned by a player didn't need to cut into more stuff that was unneeded,

DragonSlayer62 replied:
moved into malehuman.dfn

Xoduz commented on data/dfndata/items/deeds/misc_deeds.dfn:
```diff
@@ -76,4 +76,14 @@ morex=208
+
+[mannequindeed]
+{
+NAME=Mannequin Deed
```
get=base_item

Also, why is it weightless?

DragonSlayer62 replied:
fixed this and add origin to is for hs when came out

Xoduz reviewed (changes requested):
Don't forget to add the proper typescript comment headers to the scripts.

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
cleanup

DragonSlayer62 committed:
Update mannequin.js

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Update misc_deeds.dfn

DragonSlayer62 committed:
cleanup

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
cleanup

DragonSlayer62 committed:
Merge branch 'mannequin' of https://github.com/DragonSlayer62/UOX3 into mannequin

DragonSlayer62 committed:
Merge branch 'develop' into mannequin

DragonSlayer62 committed:
Update malehuman.dfn

DragonSlayer62 committed:
Changes

Fixed so no longer needed onspeech inside the manniquine to block things added a check in code

DragonSlayer62 committed:
Update mannequin.js

DragonSlayer62 committed:
Merge branch 'develop' into mannequin
